### PR TITLE
Add "geolocation task queue"

### LINKED ISSUE #80: Callbacks are invoked while in parallel
If I'm reading https://w3c.github.io/geolocation-api/#dfn-request-position correctly it goes in parallel in step 2.3 (it might be clearer to not return and say "run these steps in parallel: [substeps]") and then tries to invoke a callback in 3.1.1 at least without queuing a task.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #80

@annevk, I did as you suggested:

 1. Collapsed the "Acquire a watch id" labelled steps. 
 1. "Run these steps [=in parallel=]:". 
 1. Tied all callback invocations to a microtask.
 1. And now I return |watchId| as the final step. 

As I had to re-indent the algorithm, probably best viewed in PR-Preview.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/pull/87.html" title="Last updated on Jul 8, 2021, 4:18 AM UTC (c65844e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/87/2ae2301...c65844e.html" title="Last updated on Jul 8, 2021, 4:18 AM UTC (c65844e)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -800,7 +802,8 @@ <h2>
           {{GeolocationPositionError}} instance whose
           {{GeolocationPositionError/code}} attribute is initialized to |code|.
           </li>
-          <li>[=Invoke=] |callback| with |error|.
+          <li>[=Queue a microtask=] with step that [=invokes=] |callback| with
```
Comment (reillyeon) [inline index.html:805]:
```suggestion
          <li>[=Queue a microtask=] with a step that [=invokes=] |callback| with
```

File: index.html [hunk 2]
```diff
@@ -596,167 +596,169 @@ <h2>
...
-              |previous id|; Otherwise, let |watchId:long| be an
-              [=implementation-defined=] {{long}} that is greater than or equal
-              to zero.
-              </li>
-              <li>[=Set/Append=] |watchId| to |watchTasks|.
-              </li>
-              <li>Return |watchId| and continue [=in parallel=].
-              </li>
-            </ol>
+          <li>If |previous id| was provided, let |watchId:long| be |previous
```
Comment (reillyeon) [inline index.html:599]: This should really be `previousId`.

**Review by reillyeon — APPROVED** [review]
LGTM with some nits.

File: index.html (hunk 2 above)
Comment (marcoscaceres) [reply index.html:599]: Good suggestion, will update.

**Comment by annevk** [comment]
Pretty sure that a microtask can only be queued in the context of a task. When you're in parallel you have to queue a task.

**Comment by marcoscaceres** [comment]
@annevk, ok, I was a bit unsure. I'll queue a task, but was unsure what task queue to go with... should I mint a new one for geolocation? Or do you have a suggestion for an existing one?

**Comment by marcoscaceres** [comment]
Gave this some thought... as every browser has its own dedicated location service manager, I'm going to mint a "geolocation task source".

**Comment by marcoscaceres** [comment]
@annevk, added the new "geolocation task source" 🥫.

**Comment by annevk** [comment]
Seems reasonable. You want to use "queue a global task" and probably pass "this's relevant global object" (that you put in a variable before going in parallel).

File: index.html [hunk 3]
```diff
@@ -594,173 +594,177 @@ <h2>
...
+                      <li>Terminate this algorithm.
                       </li>
                     </ol>
                   </li>
                 </ol>
               </li>
-              <li>Clear |timerId| from the [=list of active timers=].
+              <li>Wait for a significant change of geographic position. What
+              constitutes a significant change of geographic position is left
+              to the implementation. A user agents MAY impose a rate limit on
```
Comment (reillyeon) [inline index.html:755]: Still a grammar issue here. Pick either "A user agent" or "User agents".

Comment (marcoscaceres) [inline index.html:755]:
```suggestion
              to the implementation. User agents MAY impose a rate limit on
```

**Pull request merged**